### PR TITLE
refactor: optimize `Ansi_color.strip` perf by copying substrings

### PR DESCRIPTION
**Pull request by MisterDA**
`Ansi_color.strip` removes ANSI escape sequences of colors[¹].

They start with the ASCII ESC (escape) character and end with the m character. The function used to copy valid characters character by character, but that is inefficient as we can copy to the destination buffer the whole valid substring at once.
Switching to unsafe string access in the inner loop yields another improvement.

The two combined show an improvement of 50% in a micro-benchmark realistic test case:

String of 562895 bytes with 1740 ANSI escape sequences. 
```
╭─────────────────────────┬───────────────────────────╮
│name                     │  monotonic-clock          │
├─────────────────────────┼───────────────────────────┤
│  strip strip_char       │        3090329.2703 ns/run│
│  strip strip_substring  │        1499787.6293 ns/run│
╰─────────────────────────┴───────────────────────────╯
```
[¹]: https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_(Select_Graphic_Rendition)_parameters

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Unreleased
 ----------
 
+- Optimize `Ansi_color.strip` by copying substrings without ANSI
```
Comment (rgrinberg) [inline CHANGES.md:4]: CHANGES for user facing stuff. I don't think the common user knows enough about dune's internals to care enough about this change.

**Comment by rgrinberg** [comment]
Looks good. Could we add a unit test for this function? It might require exporting some private stuff, but that's fine as long as we export it under `Private` modules.

**Comment by MisterDA** [comment]
I've added a short expect test, the function was already exported.

**Comment by rgrinberg** [comment]
Thanks. Could you add the DCO to the set second commit?

**Pull request merged**